site/content/en/docs/Getting started/first_kustomization.md

```diff
@@ -4,48 +4,189 @@ linkTitle: "Creating Your First Kustomization"
+
+```yaml
+$ kustomize build .
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: nginx
+  name: nginx
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nginx
+  name: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+        ports:
+        - containerPort: 80
+```
+
+### Customise your resources
+
+So far we have not used kustomize to do any modifications, so let's see how we can do that. Kustomize comes with a considerable number of transformers that apply changes to our manifests, and in this section we will have a look at the `namePrefix` transformer.  This transformer will add a prefix to the deployment and service names.  Modify the `kustomization.yaml` file as follows:
+
+```yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+namePrefix: example-  ### add this line
+
+resources:
+- deployment.yaml
+- service.yaml
+```
+
+After re-building we see can see our modified manifest which now has the prefixed deployment and service names:
+
+```yaml
+$ kustomize build .
+
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: nginx
+  name: example-nginx    ### service name changed here
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: nginx
+  name: example-nginx    ### deployment name changed here
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+        ports:
+        - containerPort: 80
+```
+
+### Create variants using overlays
+
+Now let's assume we need to deploy the nginx manifests from the previous section to two environments called `Staging` and `Production`. The manifests for these two environments will be mostly identical, with only a few minor changes between them. We call these two mostly identical manifests "variants". Traditionally to create variants, we could duplicate the manifests and apply the changes manually or rely on some templating engine. With kustomize, we can avoid templating and duplication of our manifests and apply the different changes we need using overlays. With this approach, the `base` would contain the common part of the our variants and the `overlays` contain our environment specific changes.
+
+Create the folder structure for the next example with bases and overlays:
+
 ```bash
+mkdir -p kustomize-example-2/base kustomize-example-2/overlays/staging kustomize-example-2/overlays/production
+cd kustomize-example-2
```

Comment:
Do you feel strongly about restarting from scratch as of this step, vs. moving the existing files they've made into a new subdir? As it is this makes me wonder if it should be a second tutorial, since it is independent of what they've done so far.

Reply:
Not really, I started from scratch to avoid having to make them undo changes and keep the name short, can think of how to update later this week

Reply:
@KnVerey consolidated into one folder